src/pudl/analysis/plant_parts_eia.py

```diff
@@ -0,0 +1,1530 @@
+
+        Returns:
+            pandas.DataFrame
+        """
+        # because lots of these input dfs include same info columns, this
+        # generates drop columnss for fuel_cost. This avoids needing to hard
+        # code columns.
+        merge_cols = ['plant_id_eia', 'generator_id', 'report_date']
+        drop_cols = [x for x in mcoe if x in gens and x not in merge_cols]
+
+        all_gens = (
+            pd.merge(
+                gens.pipe(pudl.helpers.convert_cols_dtypes, 'eia'),
+                mcoe.pipe(pudl.helpers.convert_cols_dtypes, 'eia')
+                .drop(drop_cols, axis=1),
+                on=merge_cols,
+                validate='1:1',
+                how='left'
+            )
+            .merge(
+                pudl.helpers.get_eia_ferc_acct_map(),
+                on=['technology_description', 'prime_mover_code'],
+                validate='m:1',
+                how='left'
+            )
+            .assign(installation_year=lambda x: x.operating_date.dt.year)
+            .astype({'installation_year': 'Int64'})
+            .pipe(self.label_operating_gens)
+        )
+        return all_gens
+
+    def label_operating_gens(self, gen_df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Label the operating generators.
+
+        We want to distinguish between "operating" generators (those that
+        report as "existing" and those that retire mid-year) and everything
+        else so that we can group the operating generators into their own
+        plant-parts separate from retired or proposed generators. We do this by
+        creating
+
+        This method also adds a column called "capacity_mw_eoy", which is the
+        end of year capacity of the generators. We assume that if a generator
+        isn't "existing", its EOY capacity should be zero.
+
+        Args:
+            gen_df (pandas.DataFrame): annual table of all generators from EIA.
+
+        Returns
+            pandas.DataFrame: annual table of all generators from EIA that
+            operated within each reporting year.
+
+        TODO: This function results in warning: `PerformanceWarning: DataFrame
+        is highly fragmented...` I expect this is because of the number of
+        columns that are being assigned here via `.loc[:, col_to_assign]`.
+
+        """
+        mid_year_retiree_mask = (
+            gen_df.retirement_date.dt.year == gen_df.report_date.dt.year)
+        existing_mask = (gen_df.operational_status == 'existing')
+        operating_mask = existing_mask | mid_year_retiree_mask
+        # we've going to make a new column which combines both the mid-year
+        # reitrees and the fully existing gens into one code so we can group
+        # them together later on
+        gen_df.loc[operating_mask, 'operational_status_pudl'] = 'operating'
+        gen_df.loc[~operating_mask, 'operational_status_pudl'] = (
+            gen_df.loc[~operating_mask, 'operational_status']
```

Comment:
`operating` is not a standard option for `operational_status`. the options natively are `'existing', 'retired', 'proposed'`. And I do want the retired and proposed labels because I want to keep these records seperate from each other in the aggregation process.

If there was a proposed generator and a retired generator that had similar attributes, they'd be lumped together in the later aggregations. Keeping the seperate is useful because sometimes either FERC or the depreciation studies report a retired slice of a plant in particular.